src/machine.js

```diff
@@ -60,22 +61,26 @@ class Machine {
       .map(edge => edge.getTransition().toString());
   }
 
-  getState(): string {
-    return this.#state.toString();
+  getState(): StateObject {
+    return this.#state;
```

Comment:
anything that wants to use `getState` should not need to know about the state properties- only the states... state.

I don't think it's a good idea to make this change, the StateObject's methods/properties are only for use within the state machine and the state machine must not relay this information back to the user of the state machine.

Why? Because the user of the state machine must never call the on hooks, therefore the only property the user is allowed to use is the name of the state, therefore we only return the name of the state.

Reply:
#getStateName ?